### ISSUE TITLE
Profile file names containing german Umlaute are not shown

### ISSUE THREAD

**Issue by NextTherapist** [Beep6581/RawTherapee#3092]
In my profiles I had a file called "neutral schwarzweiß.pp3", but I could not manage to choose it, not for a file and not in settings.

RT seems not to show files with ä, ö, ü or ß in their name.

[neutral schwarzweiß.pp3.txt](https://github.com/Beep6581/RawTherapee/files/89262/neutral.schwarzweiss.pp3.txt)

**Comment by TooWaBoo** [comment]
I can confirm this bug in RT 4.2.670 Gtk3, Win 10 x64

**Comment by Beep6581** [comment]
I downloaded your file and renamed it to `neutral schwarzweiß.pp3` without the `.txt`. I ran RT (Gtk3 branch), clicked on "Load a profile from file" and I see it as `neutral schwarzweiss.pp3`.
Linux, LANG=en_US.utf8

**Comment by TooWaBoo** [comment]
@Beep6581 
It's not listet in the pulldown menu, not in the Load profile dialog.

**Comment by Beep6581** [comment]
@NextTherapist please be more specific in your bug reports and don't forget to include the branch, version (with hash, e.g. 4.2.123 af354abc) and other relevant info as per http://rawpedia.rawtherapee.com/How_to_write_useful_bug_reports

**Comment by Beep6581** [comment]
Confirmed that `neutral schwarzweiß.pp3` does not show up in the processing profile drop-down.

**Comment by Hombre57** [comment]
Solved in branch gtk3-profile-panel-utf8-chars (works fine here).

I don't know why but either `Glib::ustring::compare()` or `Glib::ustring::casefold()` mix up between raw byte and chars.

**Comment by adamreichold** [comment]
@Hombre57 I basically agree to the fix, but I would have preferred to have it and the loop clean-ups in separate commits so that this will be easier to find in version control in the future. Also doesn't this affect `master` as well, so why only fix it in `gtk3`? (AFAIK there has not been a firm decision about how to proceed on the "Gtk+ situation"?)

Concerning the method of the fix, I don't think there is a [difference](https://developer.gnome.org/glibmm/stable/classGlib_1_1ustring.html#a992bf7d88a34cb50ecccc812e38348c1) between `ustring::size` and `ustring::length` as both return the number of characters. Also because this used often in the code: I think that if we know the extension we are looking for, it is much easier to just check for that instead of finding the last dot and so on, i.e.
````
size_t lastdot = currDir.find_last_of ('.');
if (lastdot != Glib::ustring::npos && lastdot == currDir.length() - 4 && currDir.substr(lastdot).casefold() == paramFileExtension) {
````
can usually be replaced by something like
````
const auto pos = currDir.size () - paramFileExtension.size ();
if (pos > 0 && currDir.substr (pos).casefold () == paramFileExtension)
````
which is simpler IMHO. (Of course `paramFileExtension` should be `.pp` and not `pp` for this to be water proof.)

**Comment by Beep6581** [comment]
Fix confirmed.

**Comment by Hombre57** [comment]
@adamreichold The only bug fix that could be ported back to Gtk2 is the grid lines bug. In this case, how should we proceed ?

Your sample code looks simpler, but is less bullet proof. In some very rare situation, where the filename is only 3 char without extension (should we really assume it will never happen ?), you code will throw a `std::out_of_range` exception because we're subtracting 2 unsigned values.

**Comment by adamreichold** [comment]
@Hombre57 So this does not affect `master`, I wasn't aware of that from the comments. 

The case where the file name equals an extension is handled by checking for the extension with the dot, i.e. `.png` instead of `png`. This will

**Comment by Hombre57** [comment]
Please forget my last comment about porting back to Gtk2, I've mixed up 2 issues I've worked on approx. at the same time. master is also affected by this bug, so I'll apply this patch to master then merge it to Gtk3.

For the `if` statement, both solutions should work and they are almost the same size. Considering that `lastdot` is used inside the block of code, we'll certainly keep the current solution.

**Comment by NextTherapist** [comment]
@Beep6581 My system ist Windows 7 64bit and it happened with the newest version of the stable branch: RT 4.2.539 for Windows Vista/7/8/10 64-bit (Branch: master), 2015-12-26 04:34 with language=system (German in my case).

**Comment by Hombre57** [comment]
@NextTherapist Bug fixed in 4.2.604.
